### DISCUSSION TITLE
Cannot access id

### DISCUSSION THREAD

**Discussion by HenrikeStahlhut** [neuefische/questions#617] [Web] [answered]
Hi everyone ✌🏻, 

**short introduction:** I have a list of all plants in my database (`AllPlantsList`). The plants are displayed as cards with their name and the room they are in. I also have a separate list for every room and the plants that are in this respective room (`PlantList`). I also have two separate collections in my database, rooms and plants. The plants collection is connected to the rooms collection by `room`, which is the `_id` of the room each plant is in. 

**Example:** 
plant collection: 
<img width="348" alt="Screenshot 2023-05-24 at 14 21 00" src="https://github.com/neuefische/questions/assets/126799897/938a415f-c216-4c30-a9c0-5d5e96d085a5">

rooms collection: 
<img width="329" alt="Screenshot 2023-05-24 at 14 20 51" src="https://github.com/neuefische/questions/assets/126799897/27ce2da0-60c4-4cf5-8a5d-4b81866924fb">

**Problem:** My problem is, that if i try to render the room in the `PlantList` in each room, everything works and name + room are displayed. If i try it with the same logic in `AllPlantList`, i get the following errors: 

![Screenshot 2023-05-24 at 14 23 35](https://github.com/neuefische/questions/assets/126799897/98f657c7-4c5a-4653-9daa-ee53958a90b2)
![Screenshot 2023-05-24 at 14 23 53](https://github.com/neuefische/questions/assets/126799897/086ce235-9dc1-4abf-8f8f-c04dc0589e08)

- I used `aggregate`, `$match` and `$lookup` to get each plants room. [Heres the code](https://github.com/HenrikeStahlhut/capstone-project/blob/main/pages/api/plants/%5Bid%5D/index.js)

- The `PlantList` gets rendered [here](https://github.com/HenrikeStahlhut/capstone-project/blob/feat/all-plants-overview/components/PlantList/PlantList.js), the `AllPlantsList` [here](https://github.com/HenrikeStahlhut/capstone-project/blob/feat/all-plants-overview/components/AllPlantsList/AllPlantsList.js). 

- [page with all plants](https://github.com/HenrikeStahlhut/capstone-project/blob/feat/all-plants-overview/pages/plants/all/index.js) 
- [page with plants in each room](https://github.com/HenrikeStahlhut/capstone-project/blob/feat/all-plants-overview/pages/rooms/%5Bid%5D.js)

I think it's because in the`[id].js` an id is given by `const router = useRouter(); const { id } = router.query;`, which is then used for fetching the api in the `PlantsList.js`. In the `AllPlantsList.js`, theres no id accessible, which results in the Error 500. 
I don't know how i can access the needed id (or whatever the problem is). 

Help would be greatly appreciated 😅 Please ask if there are any questions. 

Heres my repo: [GitHub](https://github.com/HenrikeStahlhut/capstone-project/tree/feat/all-plants-overview)

**Comment by mbosselmann** [comment, marked as answer]
Hi Henny,
you can update your `getAllPlantsFunction` and use `aggregate` and `$lookup` to get all necessary information about the plants and rooms. :)

```js
async function getAllPlants() {
  await connectDatabase();
  const elements = await Plant.aggregate([
    {
      $lookup: {
        from: "rooms",
        localField: "room",
        foreignField: "_id",
        as: "rooms",
      },
    },
  ]);
  return elements;
}
```

Your `getAllPlants` function should then send the fetch request to `api/plants`:
```js
const { data: plants, error, isLoading } = useSWR(`/api/plants`, fetcher);
```

I hope this does help. :)

One remark: I was not sure how the user can navigate to the page `plants/all`. Did I miss something or do you want to implement this later? 🤔

**Reply by HenrikeStahlhut** [reply]
@mbosselmann oh good point, i forgot to link the home button to the all plants page. Trough the Homepage, the user can later on navigate to all important pages 😄 It's a bit confusing in this state, because some pages are missing
